### PR TITLE
blog post to troubleshoot performance issue in golang

### PR DESCRIPTION
**Pull request by AndyCoveo**

### CODE REVIEW

File: _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+  bio: Principal Software Developper, Machine Learning Platform
+  twitter: Drahakar
+  image: aemond.jpg
+ 
+
+---
+
+There is an old adage that says, "Hardware eventually fails. Software eventually works." As the [Coveo Blitz](blitz.coveo.com) competition is approaching, we are ramping up the platform to receive more than 135 participants who will produce software bots that fight against one another for the prestigious Coveo Cup. One of the cool features of the platform, that we’ve created and improved over the years, is the ability to visualize past fights by downloading a replay file.
+
+But this year, like every year, was different. This year, the replays were up to 50mb large and the download time was suffering from it. By suffering I mean 50 seconds to download 40Mb; waiting for a mp3 to download in the 2000's kind of suffering. The replays are stored in S3, proxied by our servers written in go, so let's see what is going on in the go code, and maybe it will eventually work!
```
Comment (jebeaudet) [inline _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md:19]: milli bit? should be MB or MiB if you want to get fancy (same thing everywhere)

File: _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+So we’ll replace
+
+```golang
+	buff := &aws.WriteAtBuffer{}
+```
+
+with 
+```golang
+	buffer := make([]byte, 0, 1024*1000*55) //55MB
```
Comment (jebeaudet) [inline _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md:114]: jsuis tannant mais 1024 * 1000 c'est à moitié des MiB et des MB, 55*1024*1024 for 55MiB

File: _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+		return
+	}
+
+	w.Write(buffer.Bytes())
+}
+```
+
+We can see from the tracing that the time is not spent in the `getGameReplays` and there is virtually nothing after it, aside from error handling and sending the bytes of the replay to the client.
+
+A quick online search hinted that the problem could be with the http.ReponseWriter buffering the response and a call to Flush() could solve this problem, which … it did!
```
Comment (jebeaudet) [inline _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md:145]: do you have a SO thread? just a suggestion

**Review by jebeaudet — APPROVED** [review]

File: _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md (hunk 3 above)
Comment (AndyCoveo) [reply _posts/2023-01-13-improving-download-speed-of-a-s3-proxy-in-go.md:145]: I think it was unfortunatly based on multiple article i've read :(

**Pull request merged**